test/e2e/config/nutanix.yaml

```diff
@@ -0,0 +1,134 @@
+---
+# CI E2E test test configuration scenario using locally build images and manifests for:
+# - cluster-api
+# - bootstrap kubeadm
+# - control-plane kubeadm
+# - docker
+
+# For creating local dev images run make docker-build-e2e from the main CAPI repository
+
+images:
+  # Use local dev images built source tree;
+  - name: ghcr.io/nutanix-cloud-native/cluster-api-provider-nutanix/controller:e2e
+    loadBehavior: mustLoad
+  # ## PLEASE KEEP THESE UP TO DATE WITH THE COMPONENTS
+  # Cluster API v1beta1 Preloads
+  - name: registry.k8s.io/cluster-api/cluster-api-controller:v1.1.4
+    loadBehavior: tryLoad
+  - name: registry.k8s.io/cluster-api/kubeadm-bootstrap-controller:v1.1.4
+    loadBehavior: tryLoad
+  - name: registry.k8s.io/cluster-api/kubeadm-control-plane-controller:v1.1.4
+    loadBehavior: tryLoad
+
+providers:
+  - name: cluster-api
+    type: CoreProvider
+    versions:
+      - name: v1.1.4
+        # Use manifest from source files
+        value: https://github.com/kubernetes-sigs/cluster-api/releases/download/v1.1.4/core-components.yaml
+        type: "url"
+        contract: v1beta1
+        files:
+          - sourcePath: "../data/shared/metadata.yaml"
+        replacements:
+          - old: "imagePullPolicy: Always"
+            new: "imagePullPolicy: IfNotPresent"
+
+  - name: kubeadm
+    type: BootstrapProvider
+    versions:
+      - name: v1.1.4
+        # Use manifest from source files
+        value: https://github.com/kubernetes-sigs/cluster-api/releases/download/v1.1.4/bootstrap-components.yaml
+        type: "url"
+        contract: "v1beta1"
+        files:
+          - sourcePath: "../data/shared/metadata.yaml"
+        replacements:
+          - old: "imagePullPolicy: Always"
+            new: "imagePullPolicy: IfNotPresent"
+
+  - name: kubeadm
+    type: ControlPlaneProvider
+    versions:
+      - name: v1.1.4
+        # Use manifest from source files
+        value: https://github.com/kubernetes-sigs/cluster-api/releases/download/v1.1.4/control-plane-components.yaml
+        type: "url"
+        files:
+          - sourcePath: "../data/shared/metadata.yaml"
+        replacements:
+          - old: "imagePullPolicy: Always"
+            new: "imagePullPolicy: IfNotPresent"
+
+  - name: nutanix
+    type: InfrastructureProvider
+    versions:
+      - name: v0.5.0 # next; use manifest from source files
+        # type: "url"
+        # value: https://github.com/nutanix-cloud-native/cluster-api-provider-nutanix/releases/download/v0.4.0/infrastructure-components.yaml
+        type: kustomize
+        value: "../../../config/default"
+        contract: v1beta1
+        replacements:
+          - old: ghcr.io/nutanix-cloud-native/cluster-api-provider-nutanix/controller:latest
+            new: ghcr.io/nutanix-cloud-native/cluster-api-provider-nutanix/controller:e2e
+          - old: "imagePullPolicy: Always"
```

Comment:
Does this need to be: Old: Always and new IfNotPresent? No?

Reply:
its a regular expression, so both will work. also, its an optimization and we can remove it as well for running in ci